### ISSUE TITLE
flow generator not loading pretrained models nor... doing anything?

### ISSUE THREAD

**Issue by akesner1** [jbohnslav/deepethogram#100]
Hello, Iv just installed DEG and got the GUI up and running. I have three videos in my project folder and have them labeled. I am ready to train the flow generator and a couple weird things. 1) despite having all 6 of the models freshly extracted from the zip folder in my pretrained_models folder, I only see one of them (200221-11518_TinyMotionNet) in the drop down. 2) when I click train button, I error out right away. Anaconda command prompt (in run as admin) below. All I did was open the project file and hit Train in flowgenerator. I am not sure if there is something I am missing here, perhaps with the project_config.yaml file? Have I not set a path correctly?

(base) C:\WINDOWS\system32>conda activate deg2

(deg2) C:\WINDOWS\system32>deepethogram
[2022-03-22 15:59:22,586] INFO [deepethogram.gui.main.setup_gui_cfg:1250] CWD: C:\WINDOWS\system32
[2022-03-22 15:59:22,601] INFO [deepethogram.gui.main.setup_gui_cfg:1251] Configuration used: split:
  reload: true
  file: null
  train_val_test:
  - 0.8
  - 0.2
  - 0.0
compute:
  fp16: false
  num_workers: 8
  batch_size: auto
  min_batch_size: 8
  max_batch_size: 512
  distributed: false
  gpu_id: 0
  dali: false
  metrics_workers: 0
reload:
  overwrite_cfg: false
  latest: false
notes: null
log:
  level: info
run:
  type: gui
label_view_width: 31
control_arrow_jump: 31
vertical_arrow_jump: 3
cmap: deepethogram
unlabeled_alpha: 0.1
prediction_opacity: 0.2

[2022-03-22 15:59:32,997] INFO [deepethogram.gui.main.initialize_project:1002] cwd: C:\WINDOWS\system32
[2022-03-22 15:59:32,997] INFO [deepethogram.gui.main.initialize_project:1004] cwd: C:\WINDOWS\system32
[2022-03-22 15:59:33,012] INFO [deepethogram.gui.main.initialize_project:1006] loaded project configuration: split:
  reload: true
  file: null
  train_val_test:
  - 0.8
  - 0.2
  - 0.0
compute:
  fp16: false
  num_workers: 8
  batch_size: 32
  min_batch_size: 8
  max_batch_size: 512
  distributed: false
  gpu_id: 0
  dali: false
  metrics_workers: 0
reload:
  overwrite_cfg: false
  latest: false
notes: null
log:
  level: info
run:
  type: gui
  model: null
  dir: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\gui_logs\220322_155932
label_view_width: 31
control_arrow_jump: 31
vertical_arrow_jump: 3
cmap: deepethogram
unlabeled_alpha: 0.1
prediction_opacity: 0.2
postprocessor:
  type: min_bout_per_behavior
  min_bout_length: 1
augs:
  LR: 0.5
  UD: 0.0
  brightness: 0.25
  contrast: 0.1
  crop_size: null
  degrees: 10
  grayscale: 0.5
  hue: 0.1
  normalization:
    'N': 157900800
    mean:
    - 0.4593331956941742
    - 0.443834622700815
    - 0.46325002962892786
    std:
    - 0.16431895014788014
    - 0.15807391270733956
    - 0.16486905790756504
  pad: null
  random_resize: false
  resize:
  - 224
  - 224
  saturation: 0.1
project:
  class_names:
  - background
  - biting
  - boxing
  - smelling
  - escaping
  - digging
  - jumping
  - grooming
  config_file: project_config.yaml
  data_path: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\DATA
  labeler: null
  model_path: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models
  name: bbt_try2
  path: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram
  pretrained_path: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models\pretrained_models
sequence:
  filter_length: 15
train:
  loss_weight_exp: 1.0

[2022-03-22 15:59:33,012] INFO [deepethogram.gui.main.initialize_project:1007] cwd: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\gui_logs\220322_155932
[2022-03-22 15:59:33,028] INFO [deepethogram.gui.main.project_loaded_buttons:173] Number finalized labels: 3
[2022-03-22 15:59:33,616] INFO [deepethogram.gui.main.initialize_video:221] Record for loaded video: {'flow': None, 'label': 'C:\\Users\\drew\\Desktop\\BBTdata\\video_files\\bbt_try2_deepethogram\\DATA\\video_2021-03-06T21_01_15_top\\video_2021-03-06T21_01_15_top_labels.csv', 'output': None, 'rgb': 'C:\\Users\\drew\\Desktop\\BBTdata\\video_files\\bbt_try2_deepethogram\\DATA\\video_2021-03-06T21_01_15_top\\video_2021-03-06T21_01_15_top.avi', 'keypoint': None, 'key': 'video_2021-03-06T21_01_15_top'}
[2022-03-22 16:00:33,667] INFO [deepethogram.gui.main.log_idle:149] User has been idle for 60.0 seconds...
[2022-03-22 16:00:58,070] INFO [deepethogram.gui.main.get_selected_models:1118] {'flow_generator': {'no pretrained weights': None, '200221_115158_TinyMotionNet': 'C:\\Users\\drew\\Desktop\\BBTdata\\video_files\\bbt_try2_deepethogram\\models\\pretrained_models\\200221_115158_TinyMotionNet\\checkpoint.pt'}, 'feature_extractor': {'no pretrained weights': None, '200415_125824_hidden_two_stream_kinetics_degf': 'C:\\Users\\drew\\Desktop\\BBTdata\\video_files\\bbt_try2_deepethogram\\models\\pretrained_models\\200415_125824_hidden_two_stream_kinetics_degf\\checkpoint.pt'}, 'sequence': {'': None}}
[2022-03-22 16:00:58,070] INFO [deepethogram.gui.main.flow_train:338] flow_train called with args: ['python', '-m', 'deepethogram.flow_generator.train', 'project.path=C:\\Users\\drew\\Desktop\\BBTdata\\video_files\\bbt_try2_deepethogram', 'reload.weights=C:\\Users\\drew\\Desktop\\BBTdata\\video_files\\bbt_try2_deepethogram\\models\\pretrained_models\\200221_115158_TinyMotionNet\\checkpoint.pt']
[2022-03-22 16:01:01,334] INFO [__main__.flow_generator_train:54] args: C:\Users\drew\anaconda3\envs\deg2\lib\site-packages\deepethogram\flow_generator\train.py project.path=C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram reload.weights=C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models\pretrained_models\200221_115158_TinyMotionNet\checkpoint.pt
[2022-03-22 16:01:01,334] INFO [__main__.flow_generator_train:62] configuration used ~~~~~
[2022-03-22 16:01:01,334] INFO [__main__.flow_generator_train:63] split:
  reload: true
  file: null
  train_val_test:
  - 0.8
  - 0.2
  - 0.0
compute:
  fp16: false
  num_workers: 8
  batch_size: 32
  min_batch_size: 8
  max_batch_size: 512
  distributed: false
  gpu_id: 0
  dali: false
  metrics_workers: 0
reload:
  overwrite_cfg: false
  latest: false
  weights: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models\pretrained_models\200221_115158_TinyMotionNet\checkpoint.pt
notes: null
log:
  level: info
augs:
  brightness: 0.25
  contrast: 0.1
  hue: 0.1
  saturation: 0.1
  color_p: 0.5
  grayscale: 0.5
  crop_size: null
  resize:
  - 224
  - 224
  dali: false
  random_resize: false
  pad: null
  LR: 0.5
  UD: 0.0
  degrees: 10
  normalization:
    'N': 157900800
    mean:
    - 0.4593331956941742
    - 0.443834622700815
    - 0.46325002962892786
    std:
    - 0.16431895014788014
    - 0.15807391270733956
    - 0.16486905790756504
train:
  lr: 0.0001
  scheduler: plateau
  num_epochs: 10
  steps_per_epoch:
    train: 1000
    val: 200
    test: 20
  min_lr: 5.0e-07
  stopping_type: learning_rate
  milestones:
  - 50
  - 100
  - 150
  - 200
  - 250
  - 300
  weight_loss: true
  patience: 3
  early_stopping_begins: 0
  viz_metrics: true
  viz_examples: 10
  reduction_factor: 0.1
  loss_weight_exp: 1.0
  loss_gamma: 1.0
  label_smoothing: 0.05
  oversampling_exp: 0.0
  regularization:
    style: l2_sp
    alpha: 1.0e-05
    beta: 0.001
flow_generator:
  type: flow_generator
  flow_loss: MotionNet
  flow_max: 10
  input_images: 11
  flow_sparsity: false
  smooth_weight_multiplier: 1.0
  sparsity_weight: 0.0
  loss: MotionNet
  max: 5
  n_rgb: 11
  arch: TinyMotionNet
  weights: pretrained
cmap: deepethogram
control_arrow_jump: 31
label_view_width: 31
postprocessor:
  min_bout_length: 1
  type: min_bout_per_behavior
prediction_opacity: 0.2
project:
  class_names:
  - background
  - biting
  - boxing
  - smelling
  - escaping
  - digging
  - jumping
  - grooming
  config_file: project_config.yaml
  data_path: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\DATA
  labeler: null
  model_path: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models
  name: bbt_try2
  path: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram
  pretrained_path: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models\pretrained_models
run:
  type: train
  model: flow_generator
  dir: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models\220322_160101_flow_generator_train
sequence:
  filter_length: 15
unlabeled_alpha: 0.1
vertical_arrow_jump: 3

[2022-03-22 16:01:01,441] INFO [__main__.flow_generator_train:67] Total trainable params: 1,951,784
[2022-03-22 16:01:01,660] INFO [deepethogram.projects.get_weightfile_from_cfg:1053] loading pretrained weights: C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models\pretrained_models\200221_115158_TinyMotionNet\checkpoint.pt
reloading weights...
[2022-03-22 16:01:01,660] INFO [deepethogram.utils.load_state:341] loading from checkpoint file C:\Users\drew\Desktop\BBTdata\video_files\bbt_try2_deepethogram\models\pretrained_models\200221_115158_TinyMotionNet\checkpoint.pt...
[2022-03-22 16:01:01,676] INFO [__main__.get_metrics:364] key metric is SSIM
Traceback (most recent call last):
  File "C:\Users\drew\anaconda3\envs\deg2\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\drew\anaconda3\envs\deg2\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\drew\anaconda3\envs\deg2\lib\site-packages\deepethogram\flow_generator\train.py", line 374, in <module>
    flow_generator_train(cfg)
  File "C:\Users\drew\anaconda3\envs\deg2\lib\site-packages\deepethogram\flow_generator\train.py", line 76, in flow_generator_train
    lightning_module = OpticalFlowLightning(flow_generator, cfg, datasets, metrics, viz.visualize_logger_optical_flow)
  File "C:\Users\drew\anaconda3\envs\deg2\lib\site-packages\deepethogram\flow_generator\train.py", line 119, in __init__
    super().__init__(model, cfg, datasets, metrics, visualization_func)
  File "C:\Users\drew\anaconda3\envs\deg2\lib\site-packages\deepethogram\base.py", line 75, in __init__
    gpu_transforms = get_gpu_transforms(self.hparams.augs, '3d' if '3d' in arch.lower() else '2d')
  File "C:\Users\drew\anaconda3\envs\deg2\lib\site-packages\deepethogram\data\augs.py", line 205, in get_gpu_transforms
    kornia_transforms.append(K.RandomHorizontalFlip(p=augs.LR,
  File "C:\Users\drew\anaconda3\envs\deg2\lib\site-packages\kornia\augmentation\base.py", line 183, in __init__
    raise ValueError(
ValueError: `return_transform` is deprecated. Please access the transformation matrix with `.transform_matrix`. For chained matrices, please use `AugmentationSequential`.
[2022-03-22 16:01:02,108] INFO [deepethogram.gui.main.flow_train:348] Training finished. If you see error messages above, training did not complete successfully.
[2022-03-22 16:01:02,108] INFO [deepethogram.gui.main.flow_train:354] ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[2022-03-22 16:01:02,124] INFO [deepethogram.gui.main.project_loaded_buttons:173] Number finalized labels: 3

**Comment by themattinthehatt** [comment]
I just did a fresh install of deepethogram and got the exact same error (specifically the "`return_transform` is deprecated....)

**Comment by rcalfredson** [comment]
Until the code in this project is updated, looks like the training will run fine if `kornia==0.6.3` and `pytorch-lightning==1.5.10` get installed instead of the latest versions for both those two packages.

**Comment by themattinthehatt** [comment]
@rcalfredson thanks for the info! I'll give that a shot

**Comment by akesner1** [comment]
> Until the code in this project is updated, looks like the training will run fine if `kornia==0.6.3` and `pytorch-lightning==1.5.10` get installed instead of the latest versions for both those two packages.

Sorry for the naïve response, but could you advise on the commands needed to get these packages installed (and the versions that we don't want uninstalled?). Thanks!

**Comment by rcalfredson** [comment]
For sure! It would be

```
pip uninstall kornia
pip uninstall pytorch-lightning
pip install kornia==0.6.3
pip install pytorch-lightning==1.5.10
```
Hope that works all right. If you're using a virtual environment then the `pip` command should still be associated with that environment and install it there, as opposed to installing packages system-wise.

**Comment by smpaixao** [comment]
> @rcalfredson thanks for the info! I'll give that a shot

Did this work for you? I ran into your same error (specifically the "return_transform is deprecated....") but after downgrading the versions of pythorch and kornia created new problems.

Thanks for any suggestion

**Comment by jbohnslav** [comment]
Yes, this was an issue with the new version of Kornia. Thank you to @rcalfredson's for the fix here. It should be fixed as of 0aaa4d5.
